modular-docs-manual/files/TEMPLATE_ASSEMBLY_a-collection-of-modules.adoc

```diff
@@ -28,7 +28,7 @@ The `context` attribute enables module reuse. Every module ID includes {context}
 ////
 
 [role="_abstract"]
-This paragraph is the assembly introduction. It explains what the user will accomplish by working through the modules in the assembly and sets the context for the user story the assembly is based on. The text that immediately follows the `[role="_abstract"]` tag is used for search metadata.
+This paragraph is the assembly introduction. It explains what the user will accomplish by working through the modules in the assembly and sets the context for the user story the assembly is based on. The text that immediately follows the `[role="_abstract"]` tag up to a line break is used for search metadata.
```

Comment:
I don't agree with this as it is purely implementation-specific. There's no such distinction about paragraphs in AsciiDoc or Asciidoctor.

```suggestion
This paragraph is the assembly introduction. It explains what the user will accomplish by working through the modules in the assembly and sets the context for the user story the assembly is based on.
```

Reply:
> Then the solution is to indicate that this is implementation-specific, not to not provide guidance.

I don't think we should go that way. There are potentially many implementations. The standard/guidelines should be implementation-agnostic. Listing all implementation-specific details or considerations is not scalable and would only introduce clutter.

> See issue 160.

It's valid question, but I don't think there should be any hard limitation -- this should be up to the writer.

> These templates are part of FCC, which is a combination of several specific implementations.

I believe this is at the heart of this misunderstanding.

The templates are not part of FCC. The templates are a part of 'modular documentation' guidelines. FCC is a project at Red Hat that makes heavy use of modular docs principles. But these guidelines are an upstream project that can be used by anyone, regardless of what company they work for or what publication toolchain they use. Therefore, the guidelines (and templates) should not be modelled after FCC and/or Pantheon needs.

To provide an analogy: it would be like wanting to modify the DITA guidelines because a Red Hat tool asked for it.

> These templates include other items that are not "vanilla Asciidoc", including the :context: attribute.

If there are things included that are not part of vanilla AsciiDoc, then it should be fixed. However, that is not the case with the `:context:` attribute. Attributes and their use are a standard part of AsciiDoc. There's nothing non-standard in that, and all existing implementations of the AsciiDoc language support this mark-up element and translate it without problems.

Reply:
The changes @rkratky proposes still provide guidance for writing an abstract. 

@stoobie What do you mean by "this"?  If you mean `[role="_abstract"]`, it is a PV2-specific requirement. Keeping it in the mod docs templates "harms" writers who must go back and remove it to comply with their teams' requirements to *not include PV2-specific elements.* 

For example, the OpenShift Docs team has proscribed `[role="_abstract"]` to avoid its incremental inclusion in their doc set. Writers like myself who accidentally include it in their content must go back and remove it afterwards. The team's intention is to avoid "messes" that must be cleaned up when they run scripts to update their whole doc set for PV2. I expect other doc teams are also adopting this approach of avoiding incremental PV2 changes so they can efficiently migrate their content when it is time to do so.

I'll check to see if there is a definitive source for PV2 requirements like this one.

Reply:
This PV2 migration checklist covers elements that are required: https://docs.engineering.redhat.com/pages/viewpage.action?spaceKey=FCCADOPT&title=Pv2+MS3+migration+checklists